### PR TITLE
[MM-13500] Searches and loads non-visible GMs in more direct channels

### PR DESCRIPTION
**Pull request by mgdelacroix**
#### Summary
When a user is searching in the more direct channel modal, this PR adds the logic to search amongst the group channels that are not visible and loads the ones matching with the search term.

#### Ticket Link
[MM-13500](https://mattermost.atlassian.net/browse/MM-13500)

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] [Has redux changes](https://github.com/mattermost/mattermost-redux/pull/820)
- [ ] Has UI changes
- [ ] Includes text changes and localization file ([.../i18n/en.json](https://github.com/mattermost/mattermost-webapp/blob/master/i18n/en.json)) updates
- [ ] Touches critical sections of the codebase (auth, posting, etc.)

### CODE REVIEW

**Comment by jwilander** [comment]
I'm a little worried about doing this here for performance reasons. Some users who have been using Mattermost for a year or more could have a significant number of GM channels (1000+) hidden. As implemented, we would request all these users on app load which in a worst case scenario would be (7 * num of GMs). We already don't scale very well when a user has been in tons of a DMs or GMs as is, I wouldn't want to make that experience any worse.

What if we instead look at loading this profiles on demand when they're needed in the More Direct Message modals (e.g. after we search)?

**Comment by mgdelacroix** [comment]
> What if we instead look at loading these profiles on demand when they're needed in the More Direct Message modals (e.g. after we search)?

Makes sense... thanks @jwilander, I'll have a look and come back with a different approach :)

**Comment by jasonblais** [comment]
@mgdelacroix Are there next steps for this PR?

**Comment by mgdelacroix** [comment]
@jasonblais it requires some more effort than expected and got reprioritised, but it's clear what's need to be done. I'll be the next thing on my queue

File: package-lock.json [hunk 1]
```diff
@@ -10534,8 +10534,8 @@
       "integrity": "sha512-6qE4B9deFBIa9YSpOc9O0Sgc43zTeVYbgDT5veRKSlB2+ZuHNoVVxA1L/ckMUayV9Ay9y7Z/SZCLcGteW9i7bg=="
     },
     "mattermost-redux": {
-      "version": "github:mattermost/mattermost-redux#477150436ac1871d0d63954c4f4f4d16ff692483",
-      "from": "github:mattermost/mattermost-redux#477150436ac1871d0d63954c4f4f4d16ff692483",
+      "version": "github:mattermost/mattermost-redux#aa10e21b4938adf580c6be4480819008e1a33eac",
```
Comment (jespino) [inline package-lock.json]: I think this should be done after the mattermost-redux get merged into master

Comment (mgdelacroix) [reply package-lock.json]: Agree, just added this so the tests would pass

**Pull request closed without merging**